tests/Unit/PointwiseFunctions/AnalyticSolutions/WaveEquation/Test_PlaneWave.cpp

```diff
@@ -0,0 +1,145 @@
+  }
+} // namespace
+
+template <size_t Dim>
+void check_solution_y(const double kx, const double ky, const double omega,
+                      const DataVector& u,
+                      const WaveEquationSolutions::PlaneWave<Dim>& pw,
+                      const tnsr::I<DataVector, Dim, Frame::Inertial>& x,
+                      const double t) {
+  const DataVector dpsi_dy = 3.0 * ky * square(u);
+  const DataVector d2psi_dtdy = -6.0 * omega * ky * u;
+  const DataVector d2psi_dxdy = 6.0 * kx * ky * u;
+  const DataVector d2psi_dydy = 6.0 * square(ky) * u;
+  for (size_t s = 0; s < 2; ++s) {
+    CHECK(approx(dpsi_dy[s]) == pw.dpsi_dx(x, t).get(1)[s]);
+    CHECK(approx(d2psi_dtdy[s]) == pw.d2psi_dtdx(x, t).get(1)[s]);
+    CHECK(approx(d2psi_dxdy[s]) == pw.d2psi_dxdx(x, t).get(0, 1)[s]);
+    CHECK(approx(d2psi_dxdy[s]) == pw.d2psi_dxdx(x, t).get(1, 0)[s]);
+    CHECK(approx(d2psi_dydy[s]) == pw.d2psi_dxdx(x, t).get(1, 1)[s]);
+  }
+}
+
+void check_solution_z(const double kx, const double ky, const double kz,
+                      const double omega, const DataVector& u,
+                      const WaveEquationSolutions::PlaneWave<3>& pw,
+                      const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+                      const double t) {
+  const DataVector dpsi_dz = 3.0 * kz * square(u);
+  const DataVector d2psi_dtdz = -6.0 * omega * kz * u;
+  const DataVector d2psi_dxdz = 6.0 * kx * kz * u;
+  const DataVector d2psi_dydz = 6.0 * ky * kz * u;
+  const DataVector d2psi_dzdz = 6.0 * square(kz) * u;
+  for (size_t s = 0; s < 2; ++s) {
+    CHECK(approx(dpsi_dz[s]) == pw.dpsi_dx(x, t).get(2)[s]);
+    CHECK(approx(d2psi_dtdz[s]) == pw.d2psi_dtdx(x, t).get(2)[s]);
+    CHECK(approx(d2psi_dxdz[s]) == pw.d2psi_dxdx(x, t).get(0, 2)[s]);
+    CHECK(approx(d2psi_dxdz[s]) == pw.d2psi_dxdx(x, t).get(2, 0)[s]);
+    CHECK(approx(d2psi_dydz[s]) == pw.d2psi_dxdx(x, t).get(2, 1)[s]);
+    CHECK(approx(d2psi_dydz[s]) == pw.d2psi_dxdx(x, t).get(1, 2)[s]);
+    CHECK(approx(d2psi_dzdz[s]) == pw.d2psi_dxdx(x, t).get(2, 2)[s]);
+  }
+}
+
+void test_1d() {
+  const double k = 1.5;
+  const double x0 = 2.4;
+  const double omega = std::abs(k);
+  const double t = 3.1;
+  const double x1 = -0.2;
+  const double x2 = 8.7;
+  const tnsr::I<DataVector, 1, Frame::Inertial> x(DataVector({x1, x2}));
+  const DataVector u({k * (x1 - x0) - omega * t, k * (x2 - x0) - omega * t});
+  WaveEquationSolutions::PlaneWave<1> pw(
+      {{k}}, {{x0}}, std::make_unique<MathFunctions::PowX>(3));
+  check_solution_x(k, omega, u, pw, x, t);
+}
+
+void test_2d() {
+  const double kx = 1.5;
+  const double ky = -7.2;
+  const double x0 = 2.4;
+  const double y0 = -4.8;
+  const double omega = std::sqrt(square(kx) + square(ky));
+  const double t = 3.1;
+  const double x1 = -10.2;
+  const double x2 = 8.7;
+  const double y1 = -1.98;
+  const double y2 = 48.27;
+  tnsr::I<DataVector, 2, Frame::Inertial> x(DataVector({x1, x2}));
```

Comment:
I think this can be made const using the initializer_list constructor for Tensor.